susemanager-utils/susemanager-sls/modules/engines/mgr_events.py

```diff
@@ -114,45 +122,38 @@ def _connect_to_database(self):
             conn_string = "dbname='{dbname}' user='{user}' host='{host}' password='{password}'".format(
                 **db_config
             )
-        log.debug("%s: connecting to database", __name__)
+        log.debug("connecting to database")
         while True:
             try:
                 self.connection = psycopg2.connect(conn_string)
                 break
             except psycopg2.OperationalError as err:
-                log.error("%s: %s", __name__, err)
-                log.error("%s: Retrying in 5 seconds.", __name__)
+                log.error("DB Error: %s", err)
+                log.error("Retrying in 5 seconds.")
                 time.sleep(5)
         self.cursor = self.connection.cursor()
 
+    def match_additiona_events(self, tag):
+        for p in self.re_additional:
+            if p.match(tag):
+                return True
+        return False
+
     def _insert(self, tag, data):
-        self.db_keepalive()
         if (
-            any(
-                [
-                    fnmatch.fnmatch(tag, "salt/minion/*/start"),
-                    fnmatch.fnmatch(tag, "salt/job/*/ret/*"),
-                    fnmatch.fnmatch(tag, "salt/beacon/*"),
-                    fnmatch.fnmatch(
-                        tag, "salt/engines/libvirt_events/*/domain/lifecycle"
-                    ),
-                    fnmatch.fnmatch(
-                        tag, "salt/engines/libvirt_events/*/pool/lifecycle"
-                    ),
-                    fnmatch.fnmatch(
-                        tag, "salt/engines/libvirt_events/*/network/lifecycle"
-                    ),
-                    fnmatch.fnmatch(tag, "salt/engines/libvirt_events/*/pool/refresh"),
-                    fnmatch.fnmatch(tag, "salt/batch/*/start"),
-                    fnmatch.fnmatch(tag, "suse/manager/image_deployed"),
-                    fnmatch.fnmatch(tag, "suse/manager/image_synced"),
-                    fnmatch.fnmatch(tag, "suse/manager/pxe_update"),
-                    fnmatch.fnmatch(tag, "suse/systemid/generate"),
-                ]
+            # Pass all salt/job/*/ret/* events
+            self.re_job_ret.match(tag)
+            and not (
+                # Except mine updates
+                data.get("fun") == "mine.update"
+                or (
+                    # And presence test.ping returns
+                    data.get("fun") == "test.ping"
+                    and data.get("metadata", {}).get("batch-mode")
+                )
```

Comment:
Is there a particular reason behind removing the `_is_salt_mine_update`, `_is_test_ping` and `_is_batch_mode` helpers and just go with the unwrapped approach here?

Reply:
Actually these methods just adds some readability but adds complexity for the python strack, in general it's a fight for some milliseconds, but in case of having large amount of events even if they are not passing the check it could save some CPU time.